src/QUICServer.ts

```diff
@@ -10,6 +10,7 @@ import type {
   StreamReasonToCode,
 } from './types';
 import type { Header } from './native/types';
+import nodesEvents from 'events';
```

Comment:
Don't we use dom events? If so is there no setting max listeners in the regular DOM API.

Ok suppose it is a nodeism, then I reckon rather than embedding this thing here, it should be put into a utility function that you call into and you can do block it saying why this arbitrary magic number is used.

Reply:
I want to avoid hardcoded magic numbers in the main line of code.

Reply:
Ok let's investigate this concept a little more.

In the DOM EventTarget API, there's actually no such thing as max listeners.

But in node it's intended to protect from memory leaks.

The default is 10 as per `nodeEvents.defaultMaxListeners` is set to 10, and pass that, we get:

```
> (node:61422) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 foo listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
```

Now of course this does not make sense for a library like js-quic, in fact with https://github.com/MatrixAI/Polykey/issues/444 it would make even less sense.

However globally setting `nodeEvents.defaultMaxListeners = Infinity` does not make sense either, as we don't want to do global mutation from a library.

So our options are:

1. Expose the ability to configure this to directly to the `Evented` class decorator in js-events. This means using `this[eventTarget].setMaxListeners?.(X)`. Note that in TS, you may need `this[eventTarget]?.setMaxListeners?.(X)` - try it out. The `?.` will run only if the function exists. This basically does it via ducktyping. And it should be exposed as a separate utility function in js-events.
2. There are also other event targets as you are using here that is outside of the `Evented` class decorator. In those cases, you also need to potentially change the max listeners. In that case I still advocate looking for the presence of the function property, rather than importing `nodeEvents`, it is more portable in non-node environments.

I think in general this sort of thing needs to be made as a high level configuration. The `js-events` library can give you the ability to call such a thing if it existed in a node environment.

Then I think think for something like js-quic, you want this to be in the `config.ts` under `QUIConfig` type, which should default to something that represents either the ballpark figure of how many event listeners are likely to exist according to the other defaults, or set it to `Infinity`.

So I don't think these 2 lines:

```
QUICServer.ts
379:    nodesEvents.setMaxListeners(100000, this.stopAbortController.signal);

QUICSocket.ts
254:    nodesEvents.setMaxListeners(100000, this[_eventTarget]);
```

Both should be deleted in favour of a more structured way of doing this.

Reply:
@tegefaulkes create new issue out of this, and cross-reference this PR comment.

Reply:
New issue created at #105 for this.